lib/pdk/cli/set/config.rb

```diff
@@ -0,0 +1,120 @@
+module PDK::CLI
+  module Set
+    module Config
+      ALLOWED_TYPE_NAMES = %w[array boolean number string].freeze
+
+      # :nocov:
+      def self.pretty_allowed_names
+        ALLOWED_TYPE_NAMES.map { |name| "'#{name}'" }.join(', ')
+      end
+      # :nocov:
+
+      def self.transform_value(type_name, value)
+        normalized_name = type_name.downcase.strip
+        unless ALLOWED_TYPE_NAMES.include?(normalized_name)
+          raise PDK::CLI::ExitWithError, _('Unknown type %{type_name}. Expected one of %{allowed}') % { type_name: type_name, allowed: pretty_allowed_names }
+        end
+
+        # Short circuit string conversions as it's trivial
+        if normalized_name == 'string'
+          raise PDK::CLI::ExitWithError, _('An error occured converting \'%{value}\' into a %{type_name}') % { value: value.nil? ? 'nil' : value, type_name: type_name } unless value.is_a?(String)
+          return value
+        end
+
+        begin
+          case normalized_name
+          when 'array'
+            convert_to_array(value)
+          when 'boolean'
+            convert_to_boolean(value)
+          when 'number'
+            convert_to_number(value)
+          else
+            value
+          end
+        rescue ArgumentError, TypeError
+          raise PDK::CLI::ExitWithError, _('An error occured converting \'%{value}\' into a %{type_name}') % { value: value.nil? ? 'nil' : value, type_name: type_name }
+        end
+      end
+
+      def self.convert_to_array(value)
+        return [] if value.nil?
+        value.is_a?(Array) ? value : [value]
+      end
+      private_class_method :convert_to_array
+
+      def self.convert_to_boolean(value)
+        string_val = value.to_s.strip.downcase
+
+        return true  if %w[yes true -1].include?(string_val)
```

Comment:
Maybe include positive 1 here as well? I think that's more common than -1 in Unix-like environments.

Reply:
Fixed.